Updated info for lifecycle events (#544)
rudderlabs/rudderstack-docs: stream-sources/rudderstack-sdk-integration-guides/rudderstack-unity-sdk.mdx

| `configRefreshInterval` | Integer     | The SDK will fetch the config from `dashboard` after the specified time.          | `2 hours`       |
| `trackLifecycleEvents`  | Boolean | Determines if the SDK will automatically capture the application lifecycle events.                     | `true`                                                   |
| `recordScreenViews`     | Boolean | Determines if the SDK will automatically capture the screen view events.                   | `false`            |
| `autoCollectAdvertId`  | Boolean | Determines if the SDK will collect the advertisement ID.  | `false` |
| `controlPlaneUrl`       | String  | Change this parameter **only if** you are self-hosting the control plane. Check the [Self-hosted control plane](#self-hosted-control-plane) section below for more information. The SDK will add `/sourceConfig` along with this URL to fetch the source configuration. | [https://api.rudderlabs.com](https://api.rudderlabs.com) |

### Self-hosted control plane

> If you are using a device mode destination like Adjust, Firebase, etc., the Unity SDK needs to fetch the required configuration from the control plane. If you are using the <a href="https://www.rudderstack.com/docs/rudderstack-open-source/control-plane-lite/">Control Plane Lite</a> utility to host your own Control Plane, then specify `controlPlaneUrl` in your`RudderConfigBuilder` that points to your hosted source configuration file.

Reviewer: ```suggestion
If you are using a device mode destination like Adjust, Firebase, etc., the Unity SDK needs to fetch the required configuration from the control plane. If you are using the <a href="https://www.rudderstack.com/docs/rudderstack-open-source/control-plane-lite/">Control Plane Lite</a> utility to host your own Control Plane, then specify `controlPlaneUrl` in your `RudderConfigBuilder` that points to your hosted source configuration file.
```

<div class="warningBlock">
  

> You shouldn't pass the <code class="inline-code">controlPlaneUrl</code> parameter during SDK initialization if you are using the <a href="https://app.rudderstack.com">RudderStack Cloud dashboard</a>. This parameter is supported only if you are using the open source <a href="https://rudderstack.com/docs/rudderstack-open-source/control-plane-lite/">Control Plane Lite</a> utility to self-host your control plane.

Reviewer: ```suggestion
You should not pass the <code class="inline-code">controlPlaneUrl</code> parameter during the SDK initialization if you are using the <a href="https://app.rudderstack.com">RudderStack Cloud dashboard</a>. This parameter is supported only if you are using the open source <a href="https://rudderstack.com/docs/rudderstack-open-source/control-plane-lite/">Control Plane Lite</a> utility to self-host your control plane.
```

    .withAutoCollectAdvertId(true);
```

## Tracking application lifecycle events for Android platform

The Unity SDK automatically tracks the [application lifecycle events](https://www.rudderstack.com/docs/rudderstack-api/api-specification/application-lifecycle-events-spec/) to get insights into the app metrics like installs, opens, updates, etc. However, you can disable the automatic tracking by using the following code:

```csharp
RudderConfig config = new RudderConfigBuilder()
  .WithTrackLifecycleEvents(false)
  .WithDataPlaneUrl("DATA_PLANE_URL")
  .WithLogLevel(RudderLogLevel.VERBOSE)
  .Build();

> rudderClient = RudderClient.GetInstance("WRITE_KEY", config);

Reviewer 2: Let's only mention about the `  .WithTrackLifecycleEvents(false)` part. With this code snippet, the reader has to look for the change they have to make to their existing codebase.

    .withAutoCollectAdvertId(true);
```

## Tracking application lifecycle events for Android platform

> The Unity SDK tracks and sends the [application lifecycle events](https://www.rudderstack.com/docs/rudderstack-api/api-specification/application-lifecycle-events-spec/) to get insights into the app-related metrics like installs, opens, updates, etc. However, in order to track the application life cycle events on the Android platform, you need to add the `RudderPreferbs.prefab` file from the path `Assets/Rudder/RudderPreferbs.prefab` to each and every scene in your Unity app. Also, make sure that the `RudderPreferbs.prefab` is linked to the `RudderClient.cs` script.

Reviewer 2: ```suggestion
The Unity SDK tracks and sends the [application lifecycle events](https://www.rudderstack.com/docs/rudderstack-api/api-specification/application-lifecycle-events-spec/) to get insights into the app metrics like installs, opens, updates, etc. However, in order to track the application life cycle events on the Android platform, you need to add the `RudderPreferbs.prefab` file from the path `Assets/Rudder/RudderPreferbs.prefab` to every scene in your Unity app. Also, ensure that the `RudderPreferbs.prefab` is linked to the `RudderClient.cs` script.
```

Unity SDK depends on the life cycle method `onApplicationFocus` of the MonoBehaviour to determine the **Application Opened** and **Application Backgrounded** events on the Android platform. Hence, when an application is brought to focus, an **Application Opened** event is sent and 
when the application is moved out of focus, an **Application Backgrounded** event is sent. 

> So, these events might be triggered even before the Rudder SDK gets initialized so as to create actions and execute them once the SDK is initialized.

Reviewer 3: ```suggestion
So, these events might be triggered even before the Rudder SDK gets initialized so we create actions for them and execute them once the SDK is initialized.
```

```csharp
void OnApplicationFocus(bool focus) {
  Action action;

  if (trackLifeCycleEvents) {
    if (focus) {
      action = () => {
        RudderLogger.LogDebug("Tracking event Application Opened");
        Dictionary < string, object > eventProperties = new Dictionary < string, object > ();
        eventProperties.Add("from_background", fromBackGround);
        RudderMessage message = new RudderMessageBuilder().WithEventName("Application Opened").WithEventProperties(eventProperties).Build();
        _instance.Track(message);

      };
    } else {
      fromBackGround = true;
      action = () => {
        RudderLogger.LogDebug("Tracking event Application Backgrounded");
        RudderMessage message = new RudderMessageBuilder().WithEventName("Application Backgrounded").Build();
        _instance.Track(message);

      };
    }

    if (isSDKInitialized) {
      RudderLogger.LogDebug("SDK Already initialized, executing the actions directly");
      action();
    } else {
      RudderLogger.LogDebug("SDK not initialized yet, adding the actions to the list");
      actionsList.Add(action);
    }
  }

> }

Reviewer 2: We don't need to put the code in the documentation.
